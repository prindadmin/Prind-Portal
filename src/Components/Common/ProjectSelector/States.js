export const QUIESCENT = 'QUIESCENT'
export const LOADING = 'LOADING'
export const LOADING_ERROR = 'LOADING_ERROR'
export const UPDATE_ERROR = 'UPDATE_ERROR'
export const PROJECT_TYPE_SELECTOR_OPEN = 'PROJECT_TYPE_SELECTOR_OPEN'
