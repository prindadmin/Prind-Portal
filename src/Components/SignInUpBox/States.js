export const QUIESCENT = 'QUIESCENT'
export const LOADING = 'LOADING'
export const PASSWORD_RESET_SUCCESS = 'PASSWORD_RESET_SUCCESS'
export const PASSWORD_RESET_FAILED = 'PASSWORD_RESET_FAILED'
