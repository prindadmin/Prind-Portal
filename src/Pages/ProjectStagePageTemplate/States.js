export const QUIESCENT = 'QUIESCENT'

export const CONTENT_FETCH_IN_PROGRESS = 'CONTENT_FETCH_IN_PROGRESS'
export const CONTENT_FETCH_SUCCESSFUL = 'CONTENT_FETCH_SUCCESSFUL'
export const CONTENT_FETCH_FAILED = 'CONTENT_FETCH_FAILED'
