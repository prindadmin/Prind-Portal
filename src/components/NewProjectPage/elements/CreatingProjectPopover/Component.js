import React, { Component } from 'react'
//import PropTypes from 'prop-types'

import PopOverHandler from '../../../common/popOverHandler'

import {
  Spinner,
  Intent,
} from '@blueprintjs/core'

import * as strings from '../../../../data/Strings'

export class Element extends Component {
  static propTypes = {
  }

  getContent = () => {
    return (
      <React.Fragment>
        {strings.PROJECT_CREATION_BILLING_WARNING}
        <Spinner size='150' intent={Intent.DANGER}/>
      </React.Fragment>
    )
  }

  render () {
    return (
      <PopOverHandler>
        <div id='popup-greyer' onClick={(e) => e.stopPropagation()}>
          <div id='project-creation-popover'>
            <div id='popup-box'>
              <div className='project-creation-popover-container' onClick={(e) => e.stopPropagation()}>
                <div className='element-title'>
                  {strings.PROJECT_CREATION_IN_PROGRESS}
                </div>
                <div className='element-description'>
                  {this.getContent()}
                </div>
              </div>
            </div>
          </div>
        </div>
      </PopOverHandler>
    )
  }

}

export default Element
