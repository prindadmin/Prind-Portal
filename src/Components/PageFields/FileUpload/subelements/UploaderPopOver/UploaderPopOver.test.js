
import { shallow } from 'enzyme';
import Component from './UploaderPopOver'
import * as States from '../../../../../States'

const mockConfigUpdate = jest.fn();
const mockS3PutObject = jest.fn();



var props;

beforeAll(() => {

  jest.mock("aws-sdk", () => {
    return {
      config: {
        update: mockConfigUpdate
      },
      S3: jest.fn(() => ({
        putObject: mockS3PutObject
      }))
    };
  });

})

beforeEach(() => {
  props = {
    fileDetails: {
      files: [
        {
          name: "hello.csv",
          type: 'text.csv',
          size: 100,
          value: "C:\\fakepath\\hello.csv"
        }
      ]
    },
    projectID: "123",
    pageName: "design",
    fieldID: 1,
    fieldType: "file",
    onCancelPopup: jest.fn(),
    user: {
      projectS3Token: {
        AccessKeyId: "TokenAccessKeyId",
        SecretAccessKey: "TokenSecretAccessKey",
        SessionToken: "TokenSessionToken"
      }
    },
    uploadFile: jest.fn(),
    requestS3ProjectFileUploadToken: jest.fn()
  };
});


it('UploaderPopOver - Should render', () => {
  const component = shallow(<Component {...props} />);
  expect(component).toMatchSnapshot();
});


it('UploaderPopOver - Should render with missing upload token', () => {
  props.user.projectS3Token = undefined
  const component = shallow(<Component {...props} />);
  expect(component).toMatchSnapshot();
});


it('UploaderPopOver - Should render with missing upload token and then close', () => {
  props.user.projectS3Token = undefined
  const mockOnCancelPopup = jest.fn()
  props.onCancelPopup = mockOnCancelPopup
  const component = shallow(<Component {...props} />);
  const closeButton = component.find('#close-button')
  closeButton.simulate('click')
  expect(mockOnCancelPopup).toHaveBeenCalled()
});
