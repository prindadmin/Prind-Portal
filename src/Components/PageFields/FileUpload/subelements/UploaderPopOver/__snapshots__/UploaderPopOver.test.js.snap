// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UploaderPopOver - Should render 1`] = `
<PopOverHandler>
  <div
    id="popup-greyer"
  >
    <div
      id="uploader-popover"
    >
      <div
        className="error"
        id="popup-box"
      >
        <div
          className="uploader-popover-container"
        >
          <div
            className="element-title"
          >
            Upload in progress.  Please wait...
          </div>
          <div
            className="element-description"
          >
            <p>
              <b>
                File Name: 
              </b>
               
              hello.csv
            </p>
            <p>
              <b>
                Uploaded: 
              </b>
               
              0 / 100 bytes
            </p>
          </div>
          <Blueprint3.ProgressBar
            intent="danger"
            value={0}
          />
          <div>
            <div>
              <b>
                ERROR: There was an error uploading the file, please try again
              </b>
            </div>
            <div>
              <Blueprint3.Button
                id="close-button"
                intent="danger"
                onClick={[Function]}
                text="Close"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</PopOverHandler>
`;

exports[`UploaderPopOver - Should render with missing upload token 1`] = `
<PopOverHandler>
  <div
    id="popup-greyer"
  >
    <div
      id="uploader-popover"
    >
      <div
        className="error"
        id="popup-box"
      >
        <div
          className="uploader-popover-container"
        >
          <div
            className="element-title"
          >
            Upload in progress.  Please wait...
          </div>
          <div
            className="element-description"
          >
            <p>
              <b>
                File Name: 
              </b>
               
              hello.csv
            </p>
            <p>
              <b>
                Uploaded: 
              </b>
               
              0 / 100 bytes
            </p>
          </div>
          <Blueprint3.ProgressBar
            intent="danger"
            value={0}
          />
          <div>
            <div>
              <b>
                ERROR: There was an error uploading the file, please try again
              </b>
            </div>
            <div>
              <Blueprint3.Button
                id="close-button"
                intent="danger"
                onClick={[Function]}
                text="Close"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</PopOverHandler>
`;
